common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -677,94 +677,92 @@ bool mon_wlan_hal_dwpal::update_vap_stats(const std::string &vap_iface_name, SVa
 bool mon_wlan_hal_dwpal::update_stations_stats(const std::string &vap_iface_name,
                                                const std::string &sta_mac, SStaStats &sta_stats)
 {
-    char *reply = nullptr;
+    const char *tmp_str;
+    int64_t tmp_int;
+    parsed_line_t reply;
 
     std::string cmd = "GET_STA_MEASUREMENTS " + vap_iface_name + " " + sta_mac;
 
-    if (!dwpal_send_cmd(cmd, &reply)) {
+    if (!dwpal_send_cmd(cmd, reply)) {
         LOG(ERROR) << __func__ << " failed";
         return false;
     }
 
-    size_t numOfValidArgs[10] = {0}, replyLen = strnlen(reply, HOSTAPD_TO_DWPAL_MSG_LENGTH);
-    uint64_t BytesSent = 0, BytesReceived = 0, PacketsSent = 0, PacketsReceived = 0,
-             LastDataDownlinkRate = 0, LastDataUplinkRate = 0, Active = 0;
-    char ShortTermRSSIAverage[32] = {'\0'};
-    char SNR[32]                  = {'\0'};
-    FieldsToParse fieldsToParse[] = {
-        {(void *)&BytesSent, &numOfValidArgs[0], DWPAL_LONG_LONG_INT_PARAM, "BytesSent=", 0},
-        {(void *)&BytesReceived, &numOfValidArgs[1], DWPAL_LONG_LONG_INT_PARAM,
-         "BytesReceived=", 0},
-        {(void *)&PacketsSent, &numOfValidArgs[2], DWPAL_LONG_LONG_INT_PARAM, "PacketsSent=", 0},
-        {(void *)&PacketsReceived, &numOfValidArgs[3], DWPAL_LONG_LONG_INT_PARAM,
-         "PacketsReceived=", 0},
-        {(void *)&sta_stats.retrans_count, &numOfValidArgs[4], DWPAL_INT_PARAM, "RetransCount=", 0},
-        {(void *)ShortTermRSSIAverage, &numOfValidArgs[5], DWPAL_STR_PARAM,
-         "ShortTermRSSIAverage=", sizeof(ShortTermRSSIAverage)},
-        {(void *)SNR, &numOfValidArgs[6], DWPAL_STR_PARAM, "SNR=", sizeof(SNR)},
-        {(void *)&Active, &numOfValidArgs[7], DWPAL_LONG_LONG_INT_PARAM, "Active=", 0},
-        {(void *)&LastDataDownlinkRate, &numOfValidArgs[8], DWPAL_LONG_LONG_INT_PARAM,
-         "LastDataDownlinkRate=", 0},
-        {(void *)&LastDataUplinkRate, &numOfValidArgs[9], DWPAL_LONG_LONG_INT_PARAM,
-         "LastDataUplinkRate=", 0},
-
-        /* Must be at the end */
-        {NULL, NULL, DWPAL_NUM_OF_PARSING_TYPES, NULL, 0}};
-
-    if (dwpal_string_to_struct_parse(reply, replyLen, fieldsToParse, sizeof(SStaStats)) ==
-        DWPAL_FAILURE) {
-        LOG(ERROR) << "DWPAL parse error ==> Abort";
+    // RSSI
+    if (!read_param_str("ShortTermRSSIAverage", reply, &tmp_str)) {
```

Comment:
Why not have a single `read_param` (or `parse_param`) that accepts different variable types (string, int etc.)?